src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.RegionBuilder.cs

```diff
@@ -252,27 +229,28 @@ public void Free()
                 LastBlock = null;
                 Regions?.Free();
                 Regions = null;
-                Methods?.Free();
-                Methods = null;
+                LocalFunctions?.Free();
+                LocalFunctions = null;
             }
 
             public ControlFlowRegion ToImmutableRegionAndFree(ArrayBuilder<BasicBlockBuilder> blocks,
-                                                                    ArrayBuilder<IMethodSymbol> methods,
-                                                                    ImmutableDictionary<IMethodSymbol, (ControlFlowRegion region, IOperation operation, int ordinal)>.Builder methodsMap,
-                                                                    ControlFlowRegion enclosing)
+                                                              ArrayBuilder<IMethodSymbol> localFunctions,
+                                                              ImmutableDictionary<IMethodSymbol, (ControlFlowRegion region, ILocalFunctionOperation operation, int ordinal)>.Builder localFunctionsMap,
+                                                              ImmutableDictionary<IFlowAnonymousFunctionOperation, (ControlFlowRegion, int)>.Builder anonymousFunctionsMapOpt,
+                                                              ControlFlowRegion enclosing)
             {
 #if DEBUG
                 Debug.Assert(!_aboutToFree);
 #endif 
                 Debug.Assert(!IsEmpty);
 
-                int methodsBefore = methods.Count;
+                int localFunctionsBefore = localFunctions.Count;
 
-                if (HasMethods)
+                if (HasLocalFunctions)
```

Comment:
>if (HasLocalFunctions) [](start = 16, length = 22)

What's the advantage of wrapping the loop with a conditional check? (same question applies on line 280 below)

Reply:
> What's the advantage of wrapping the loop with a conditional check? (same question applies on line 280 below)

The advantage is in not crashing, LocalFunctions is lazy, could be null.

---
In reply to: [193833788](https://github.com/dotnet/roslyn/pull/27536#discussion_r193833788) [](ancestors = 193833788)

Reply:
That's a good advantage ;-)
Thanks

---
In reply to: [193839374](https://github.com/dotnet/roslyn/pull/27536#discussion_r193839374) [](ancestors = 193839374,193833788)